### DISCUSSION TITLE
can not build kong-ingress-controller v2.3.1

### DISCUSSION THREAD

**Discussion by albertforweb** [Kong/kubernetes-ingress-controller#3798] [General]
Hi experts,

I downloaded kong ingress controller v2.3.1  from https://github.com/Kong/kubernetes-ingress-controller/archive/refs/tags/v2.3.1.tar.gz

and extracted it to my go folder at /go/src/github.com/kong/kubernetes-ingress-controller
then run "make build" command, but not be able to build it. see below:

```
/go/src/github.com/kong/kubernetes-ingress-controller# make build
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
go build -a -o bin/manager -ldflags "-s -w \
	-X github.com/kong/kubernetes-ingress-controller/v2/internal/metadata.Release= \
	-X github.com/kong/kubernetes-ingress-controller/v2/internal/metadata.Commit= \
	-X github.com/kong/kubernetes-ingress-controller/v2/internal/metadata.Repo=" internal/cmd/main.go
# sigs.k8s.io/json/internal/golang/encoding/json
/go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
/go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
Makefile:75: recipe for target 'build' failed
make: *** [build] Error 2
```

my golang version : 

```
# go version
go version go1.16.7 linux/amd64
```

how can I build this version to generate the kong-ingress-controller binary ? 

Thanks

**Comment by pmalek** [comment]
Hi @albertforweb 👋 

I've just downloaded this tar ball and it seems that it compiles just fine for me. The one issue I have with it is that the linter run wouldn't pass and hence block the `build` makefile target to get to the compilation stage. You can workaround that by commenting out the lint step as in 

```
.PHONY: build
build: generate #fmt vet lint
	go build -a -o bin/manager -ldflags "-s -w \
		-X github.com/kong/kubernetes-ingress-controller/v2/internal/metadata.Release=$(TAG) \
		-X github.com/kong/kubernetes-ingress-controller/v2/internal/metadata.Commit=$(COMMIT) \
		-X github.com/kong/kubernetes-ingress-controller/v2/internal/metadata.Repo=$(REPO_INFO)" internal/cmd/main.go
``` 

in the `Makefile`.

I've compiled it using go `1.20.2`. You might need to update your toolchain to succeed compiling that version because already in 2.3 we've specified go 1.18 in `go.mod`

```
module github.com/kong/kubernetes-ingress-controller/v2

go 1.18
...
```

I'm going to close this discussion for now but please feel free to reach out if you encounter any further issues.
